corehq/apps/userreports/tasks.py

```diff
@@ -80,15 +80,21 @@ def rebuild_indicators(indicator_config_id):
         redis_key = _get_redis_key_for_config(config)
 
     adapter.rebuild_table()
-    relevant_ids = get_doc_ids_in_domain_by_type(
-        config.domain,
-        config.referenced_doc_type,
-        database=couchdb,
-    )
-    for docs in chunked(relevant_ids, 1000):
-        redis_client.sadd(redis_key, *docs)
-
-    _build_indicators(indicator_config_id, relevant_ids)
+    relevant_ids_chunk = []
+    for relevant_id in iterate_doc_ids_in_domain_by_type(
```

Comment:
could doc id's get skipped if documents are deleted concurrently?  Suppose you have documents A, B, C, D and pages [A, B], [C, D].  The first page [A, B] is read.  Then document B is deleted.  The next page would be [D] and C would never be read.

Not sure if this mental model is correct, but seems plausible.